app/src/main/java/org/oppia/app/customview/CustomProgressView.kt

```diff
@@ -0,0 +1,119 @@
+package org.oppia.app.customview
+
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.graphics.RectF
+import android.util.AttributeSet
+import android.util.TypedValue
+import android.view.View
+import androidx.core.content.ContextCompat
+import org.oppia.app.R
+
+private const val STROKE_DASH_GAP_IN_DEGREE = 12
+
+/**
+ * CustomView to represent story progress in Topic-Play-Tab.
+ * Without chaptersFinished and totalChapters values this custom-view cannot be created.
+ *
+ * Reference: // https://stackoverflow.com/a/39210676
+ */
+class CustomProgressView : View {
```

Comment:
Can we come up with a more contextual name than 'custom'? Maybe something like 'SegmentedCircularProgressView'.